primitives/sidechain-mc-hash/src/lib.rs

```diff
@@ -59,32 +57,80 @@ impl Deref for McHashInherentDataProvider {
 	}
 }
 
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize))]
+pub struct MainchainBlock {
+	pub number: McBlockNumber,
+	pub hash: McBlockHash,
+	pub epoch: McEpochNumber,
+	pub slot: McSlotNumber,
+	pub timestamp: u64, // seconds since UNIX_EPOCH
+}
+
+/// Queries about Cardano Blocks
+#[async_trait]
+pub trait McHashDataSource {
+	type Error;
```

Comment:
I see quite complex `&(dyn McHashDataSource<Error = E> + Send + Sync)` because of this.
What would we lose, in terms of error handling, if concrete error with message field was used here instead?

Reply:
That might be an over-engineering on my part tbh. I didn't like that either the trait here must anticipate what kinds of error the data source might return, or use an opaque `Box<dyn Error>`, which is used anyway further down the execution flow. I'll try and see if switching this to a concrete type has any real drawbacks here.

Reply:
I changed it to use boxed error, we don't really loose anything here, since we're wrapping the error later anyway. I will do the same change for the native token IDP in a separate PR

Reply:
https://github.com/input-output-hk/partner-chains/pull/153